### ISSUE TITLE
Grid Row Actions

### ISSUE THREAD

**Issue by willishq** [query-grid/query-grid#7]
We need to be able to add `actions` to the grid, to be appended to each row.

I imagine it working like this:

`$grid->addRowAction('view', '/view/url/for/row')`
`$grid->addRowAction('edit', '/edit/url/for/row')`
`$grid->addRowAction('delete', '/delete/url/for/row')`

maybe the url for the row action should be a callback which gets passed the row.

When rendering the rowAction, it should also have a callback which returns the html.

```php
echo $grid->renderRowAction('view', function ($url, $row) {
    return "<a href=\"{$url}\">View</a>";
});
echo $grid->renderRowAction('edit', function ($url, $row) {
    return "<a href=\"{$url}\">Edit</a>";
});
echo $grid->renderRowAction('view', function ($url, $row) {
    return "<form action=\"$url\" method="POST"><button type=\"submit\">Delete</button></form>";
});

**Comment by willishq** [comment]
I'm considering whether this is actually worth doing. The more I try to make the functionality, The more I think it would be easier to just create the buttons manually, rather than actually implement them this way.

If we create UI integrations for frameworks, maybe it'd be nice to add to the framework implementation, however the core package doesn't really need to know about them, and anything we can do to implement them I fear will just be bloat.

**Comment by maxalmonte14** [comment]
I don't think this should be a core function, maybe the user wants to implement an action handled by Javascript or something else; also the package doesn't mix HTML and PHP until now, I think is better to keep it that way.
